### PR TITLE
Prove claims in fixed order

### LINKED ISSUE #2184: Prove claims in fixed order
The backend is not consistent about the order in which claims are proved. Claims should be proved in the order they are written in the specification.

### PR DESCRIPTION
**Pull request by andreiburdusa**
---
Fixes #2184 
###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`

### CODE REVIEW

File: kore/app/share/GlobalMain.hs [hunk 1]
```diff
@@ -518,8 +526,15 @@ type LoadedDefinition = (Map ModuleName LoadedModule, Map Text AstLocation)
...
-    =<< traverse parseDefinition filePaths
+    loadedDefinitions <&> sortClaims
+  where
+    loadedDefinitions =
+        Monad.foldM verifyDefinitionWithBase mempty
+        =<< traverse parseDefinition filePaths
+
+    sortClaims :: LoadedDefinition -> LoadedDefinition
+    sortClaims modulePair =
+        modulePair & Lens._1 . Lens.traversed %~ sortModuleClaims
```
Comment (MirceaS) [inline kore/app/share/GlobalMain.hs:537]:
```suggestion
    sortClaims = Lens._1 . Lens.traversed %~ sortModuleClaims
```
Won't this be simpler like this?

File: kore/src/Kore/ASTVerifier/DefinitionVerifier.hs [hunk 2]
```diff
@@ -138,3 +148,11 @@ verifyAndIndexDefinitionWithBase
...
   where
     modulesByName = Map.fromList . map (\m -> (moduleName m, m))
+
+sortModuleClaims
+    :: VerifiedModule declAtts
+    -> VerifiedModule declAtts
+sortModuleClaims verifiedModule =
+    verifiedModule
+    & field @"indexedModuleClaims"
+        %~ sortBy (on compare (Attribute.sourceLocation . fst))
```
Comment (ttuegel) [inline kore/src/Kore/ASTVerifier/DefinitionVerifier.hs:158]: We can use `comparing` from `Data.Ord`:

```suggestion
        %~ sortBy (comparing (Attribute.sourceLocation . fst))
```

File: kore/test/Test/Kore/Builtin.hs [hunk 3]
```diff
@@ -151,3 +174,25 @@ notInternalizes name origin =
 
 metadata :: SmtMetadataTools Attribute.Symbol
 metadata = Builtin.testMetadataTools
+
+test_SortClaims :: TestTree
+test_SortClaims =
```
Comment (ttuegel) [inline kore/test/Test/Kore/Builtin.hs:179]: I would suggest having the test name match the function name here:

```suggestion
test_sortModuleClaims :: TestTree
test_sortModuleClaims =
```

File: kore/test/Test/Kore/Builtin/Definition.hs [hunk 4]
```diff
@@ -1747,6 +1751,46 @@ testModule =
...
+        { moduleName = testModuleName
+        , moduleAttributes = Attributes []
+        , moduleSentences =
+            [ SentenceClaimSentence . SentenceClaim $
+                (SentenceAxiom
+                    { sentenceAxiomParameters = [SortVariable (testId "sv1")]
+                    , sentenceAxiomPattern =
+                        asParsedPattern
+                            $ PatternF.StringLiteralF
+                            $ Const (StringLiteral "a")
```
Comment (ttuegel) [inline kore/test/Test/Kore/Builtin/Definition.hs:1766]: You can also make a `TermLike` here with the convenient `mk*` constructors and call `Kore.Builtin.External.externalize`.

**Review by ttuegel — APPROVED** [review]

**Pull request merged**